packages/editor/src/pages/Editor/store/settings/utilities.ts

```diff
@@ -20,19 +22,25 @@ export const defaultSettings: ISettings = {
   'editor.wordWrap': 'bounded',
 };
 
+export const invisibleDefaultSettings: { [key: string]: any } = {};
+
 const getTabSize = (userSettings: Partial<ISettings>): number =>
   userSettings && userSettings['editor.tabSize']
     ? userSettings['editor.tabSize']!
     : defaultSettings['editor.tabSize'];
 
 const getDefaultSettingsContent = (userSettings: Partial<ISettings>): string => {
   const tabSize = getTabSize(userSettings);
-  return JSON.stringify(defaultSettings, null, tabSize) + '\n';
+  return YAML.safeDump(defaultSettings, { indent: tabSize });
 };
 
 const getUserSettingsContent = (userSettings: Partial<ISettings>): string => {
+  if (Object.keys(userSettings).length === 0) {
+    return '';
+  }
+
   const tabSize = getTabSize(userSettings);
-  return JSON.stringify(userSettings, null, tabSize) + '\n';
```

Comment:
no new line at the end of the file?

Reply:
I think js-yaml is automatically adding one.  When I had the `+ '\n'` I was getting two newlines...